packages/flutter/lib/src/material/text_field.dart

```diff
@@ -754,15 +722,37 @@ class _TextFieldState extends State<TextField> with AutomaticKeepAliveClientMixi
 
     bool forcePressEnabled;
     TextSelectionControls textSelectionControls;
+    bool paintCursorAboveText;
+    bool cursorOpacityAnimates;
+    Offset cursorOffset;
```

Comment:
Do we not need the default value of 2 coming from the widget.cursorOffset?

Some random yak shaving while we're here. I had to click through all the editables to realize that RenderEditable handles nulls. Can we add it to the doc?

Reply:
I also thing I'm leaning towards liking the getters more. It takes less effort to read and realize that we're only dealing with an Android and an iOS value and don't have to follow the local variable through to watch out for the fallthrough/null case.

Reply:
The cursorOffset is only defined for iOS. 

I moved the couple of getters that were returning platform-specific values to the block of code that handles configuring the EditableText's other platform-specific values because it's easier to read (all in one place), simpler, and that's the way the rest of it already was.

I'll be happy to add some documentation about null parameters to RenderEditable in a separate PR.

Reply:
oops, sorry, there is no widget.cursorOffset. I misread widget.cursorWidth